lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/IMRUDriver.cs

```diff
@@ -629,16 +635,12 @@ public void OnNext(IFailedTask failedTask)
             {
                 using (Logger.LogFunction("IMRUDriver::IFailedTask"))
                 {
-                    if (_taskManager.AreAllTasksCompleted())
-                    {
-                        Logger.Log(Level.Info,
-                            "Task with Id: {0} failed but all IMRU tasks are completed. So ignoring.",
-                            failedTask.Id);
-                        return;
-                    }
-
                     switch (_systemState.CurrentState)
                     {
+                        case SystemState.TasksCompleted:
+                            _taskManager.RecordFailedTaskDuringRunningOrSubmissionState(failedTask);
```

Comment:
Unlike ShutDown state, when we change system state to TasksCompleted, we don't close all the running tasks and change running tasks' state from running to TaskWaitingForClose. So if we receive   any FailedTask after TasksCompleted, most possibly the task was Running.  RecordFailedTaskDuringRunningOrSubmissionState method will remove the task from _runningTasks if it exists while RecordFailedTaskDuringSystemShuttingDownState assume the task must be not running. 
For task state, the original task state should be not TaskFailedByEvaluatorFailure, if yes, there was a unignored failed evaluator which happened before the TasksCompleted, then the system should be in shutdown state. 
Similarly, the task shouldn't in waitingForClose state, as we only enforce to close tasks after the system state is set to shutdown. 
Therefore we should call RecordFailedTaskDuringRunningOrSubmissionState